app/controllers/application_controller.rb

```diff
@@ -1,7 +1,41 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :set_locale
+
+  def set_locale
+    if user_signed_in? && !current_user.language.blank?
```

Comment:
Or could avoid checking `current_user.language` entirely and set it unconditionally and use `||=` below? Up to you.